app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/DummyDatabaseManagement.kt

```diff
@@ -0,0 +1,145 @@
+package com.github.HumanLearning2021.HumanLearningApp.model
+
+import android.net.Uri
+import android.os.Parcelable
+import com.google.common.collect.ImmutableSet
+import kotlinx.parcelize.Parcelize
+import java.lang.IllegalArgumentException
+
+/**
+ * Dummy implementation of a database manager
+ * Dataset & category names and ids are equivalent
+ */
+data class DummyDatabaseManagement(val databaseService: DummyDatabaseService): DatabaseManagement {
+    companion object {
+        val staticDummyDatabaseManagement = DummyDatabaseManagement(DummyDatabaseService())
+    }
+
+    override suspend fun getPicture(category: Category): CategorizedPicture? {
+        return try {
+            databaseService.getPicture(category)
+        } catch (e: IllegalArgumentException) {
+            throw e
+        }
+    }
+
+    override suspend fun putPicture(picture: Uri, category: Category): CategorizedPicture {
+       return try {
+           databaseService.putPicture(picture, category)
+       } catch (e: IllegalArgumentException) {
+           throw e
+       }
+    }
+
+    override suspend fun getCategoryById(categoryId: Any): Category? {
+        return databaseService.getCategory(categoryId)
+    }
+
+    override suspend fun getCategoryByName(categoryName: String): Collection<Category> {
+        val categories = databaseService.getCategories()
+        val res: MutableSet<Category> = mutableSetOf()
+        for (c in categories) {
+            if (c.name == categoryName) {
+                res.add(c)
+            }
+        }
+        return ImmutableSet.copyOf(res)
```

Comment:
maybe consider factoring this out (as suggested by codeclimate)

Reply:
I think it is too complex to make it into a useful method to warrant the effort